src/comparator/message_comparator.py

```diff
@@ -3,62 +3,91 @@
-            self._compareNestedMessages(message_original.nested_types_by_name, message_update.nested_types_by_name)
+        # Descriptor.nested_types_by_name (dict str -> Descriptor)
+        # indexed by name. Recursively call _compare for nested
+        # message type comparison.
+        if message_original.nested_type or message_update.nested_type:
+            self._compareNestedMessages(
+                {m.name: m for m in message_original.nested_type},
+                {m.name: m for m in message_update.nested_type})
 
-        # 5. TODO(xiaozhenliu): check `google.api.resource` annotation.     
+        # 5. TODO(xiaozhenliu): check `google.api.resource` annotation.
 
     def _compareNestedFields(self, fields_dict_original, fields_dict_update):
-        fields_unique_original = list(set(fields_dict_original.keys()) - set(fields_dict_update.keys()))
-        fields_unique_update = list(set(fields_dict_update.keys()) - set(fields_dict_original.keys()))
-        fieldsIntersaction = list(set(fields_dict_original.keys()) & set(fields_dict_update.keys()))
-        
+        fields_unique_original = list(
+            set(fields_dict_original.keys()) - set(
+                fields_dict_update.keys()))
+        fields_unique_update = list(
+            set(fields_dict_update.keys()) - set(
+                fields_dict_original.keys()))
+        fieldsIntersaction = list(
+            set(fields_dict_original.keys()) & set(fields_dict_update.keys()))
+
         for fieldNumber in fields_unique_original:
             FieldComparator(fields_dict_original[fieldNumber], None).compare()
         for fieldNumber in fields_unique_update:
             FieldComparator(None, fields_dict_update[fieldNumber]).compare()
         for fieldNumber in fieldsIntersaction:
-            FieldComparator(fields_dict_original[fieldNumber], fields_dict_update[fieldNumber]).compare()
-    
-    def _compareNestedMessages(self, nested_msg_dict_original, nested_msg_dict_update):
-        msg_unique_original = list(set(nested_msg_dict_original.keys()) - set(nested_msg_dict_update.keys()))
-        msg_unique_update = list(set(nested_msg_dict_update.keys()) - set(nested_msg_dict_original.keys()))
-        msgIntersaction = list(set(nested_msg_dict_original.keys()) & set(nested_msg_dict_update.keys()))
+            FieldComparator(
+                fields_dict_original[fieldNumber],
+                fields_dict_update[fieldNumber]).compare()
+
+    def _compareNestedMessages(
+            self,
+            nested_msg_dict_original,
+            nested_msg_dict_update
+    ):
+        msg_unique_original = list(
+            set(nested_msg_dict_original.keys()) - set(nested_msg_dict_update.keys()))
+        msg_unique_update = list(
+            set(nested_msg_dict_update.keys()) - set(nested_msg_dict_original.keys()))
+        msgIntersaction = list(
+            set(nested_msg_dict_original.keys()) & set(nested_msg_dict_update.keys()))
```

Comment:
Nit: since we're using the same two sets three times each, why not lift them into variables?
```python
original = set(nested_msg_dict_original)
update = set(nested_msg_dict_update)
msg_unique_original = original - update
msg_unique_update = update - original
msg_intersection = original & update
```

Reply:
Sounds reasonable, also modified this in #2